patches/server/0968-New-event-verbose-logging-command.patch

```diff
@@ -0,0 +1,264 @@
+ 
+         HandlerList handlers = event.getHandlers();
+         RegisteredListener[] listeners = handlers.getRegisteredListeners();
+diff --git a/src/main/java/io/papermc/paper/util/misc/EventWrapper.java b/src/main/java/io/papermc/paper/util/misc/EventWrapper.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..9ad6f1ec5f8e9799d9dfb6e2d73420163e735f08
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/util/misc/EventWrapper.java
+@@ -0,0 +1,66 @@
++package io.papermc.paper.util.misc;
++
++import com.google.gson.JsonObject;
++import net.minecraft.server.MinecraftServer;
++import org.bukkit.event.Event;
++import org.checkerframework.checker.nullness.qual.Nullable;
++
++import java.lang.reflect.Field;
++import java.lang.reflect.Modifier;
++import java.time.LocalDateTime;
++import java.time.format.DateTimeFormatter;
++import java.util.ArrayList;
++import java.util.Arrays;
++import java.util.List;
++
++/**
++ * A simple event wrapper with a timestamp to allow for QoL JSon serialization
++ */
++public class EventWrapper {
++
++    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd_HH.mm.ss");
++    private final String timestamp;
++    private final Event event;
++
++    public EventWrapper(Event event) {
++        this.timestamp = FORMATTER.format(LocalDateTime.now());
++        this.event = event;
++    }
++
++    /**
++     * Properly serialize our Wrapper into a JsonObject, so we can log it properly
++     *
++     * @return Null if {@link #shouldSerialize(List)} returns false, otherwise it's your object
++     */
++    @Nullable
++    public JsonObject serialize() {
++        JsonObject parent = new JsonObject();
++        parent.addProperty("timestamp", timestamp);
++        JsonObject obj = new JsonObject();
++        obj.addProperty("event", event.getEventName());
++        try {
++            List<Field> fields = new ArrayList<>(Arrays.asList(event.getClass().getDeclaredFields()));
++            Class<?> clazz = event.getClass().getSuperclass();
++            while (clazz != null) {
++                fields.addAll(Arrays.asList(clazz.getDeclaredFields()));
++                clazz = clazz.getSuperclass();
++            }
++            if (!shouldSerialize(fields)) return null; //check current event class fields if we should serialize
++            for (Field f : fields) {
++                if (Modifier.isStatic(f.getModifiers())) continue;
++                f.setAccessible(true);
++                Object field = f.get(event);
++                obj.addProperty(f.getName(), field == null ? "null" : field.toString());
++            }
++        } catch (SecurityException | IllegalAccessException ex) {
++            MinecraftServer.LOGGER.warn("Failure to serialize event " + event.getEventName(), ex);
++        }
++        parent.add("info", obj);
++        return parent;
++    }
++
++    public boolean shouldSerialize(List<Field> fields) {
```

Comment:
Exclude the name field in the event class, it causes duplicate names to be logged.